services/manifest-repo-export-service/pkg/service/git_test.go

```diff
@@ -0,0 +1,1140 @@
+					return err
+				}
+				err = dbHandler.DBInsertRelease(ctx, transaction, db.DBReleaseWithMetaData{
+					EslVersion:    db.InitialEslVersion,
+					ReleaseNumber: 2,
+					Created:       time.Time{},
+					App:           "app",
+					Manifests:     db.DBReleaseManifests{},
+					Metadata:      db.DBReleaseMetaData{},
+				}, 1)
+				if err != nil {
+					return err
+				}
+				err = dbHandler.DBInsertRelease(ctx, transaction, db.DBReleaseWithMetaData{
+					EslVersion:    db.InitialEslVersion,
+					ReleaseNumber: 3,
+					Created:       time.Time{},
+					App:           "app",
+					Manifests:     db.DBReleaseManifests{},
+					Metadata:      db.DBReleaseMetaData{},
+				}, 2)
+				if err != nil {
+					return err
+				}
+				err = dbHandler.DBInsertRelease(ctx, transaction, db.DBReleaseWithMetaData{
+					EslVersion:    db.InitialEslVersion,
+					ReleaseNumber: 1,
+					Created:       time.Time{},
+					App:           "app-1",
+					Manifests:     db.DBReleaseManifests{},
+					Metadata:      db.DBReleaseMetaData{},
+				}, 3)
+				if err != nil {
+					return err
+				}
+				err = dbHandler.DBInsertRelease(ctx, transaction, db.DBReleaseWithMetaData{
+					EslVersion:    db.InitialEslVersion,
+					ReleaseNumber: 1,
+					Created:       time.Time{},
+					App:           "app-2",
+					Manifests:     db.DBReleaseManifests{},
+					Metadata:      db.DBReleaseMetaData{},
+				}, 4)
+				if err != nil {
+					return err
+				}
+				err = dbHandler.DBInsertRelease(ctx, transaction, db.DBReleaseWithMetaData{
+					EslVersion:    db.InitialEslVersion,
+					ReleaseNumber: 1,
+					Created:       time.Time{},
+					App:           "app-3",
+					Manifests:     db.DBReleaseManifests{},
+					Metadata:      db.DBReleaseMetaData{},
+				}, 5)
+				if err != nil {
+					return err
+				}
+				err = dbHandler.DBInsertAllReleases(ctx, transaction, "app", []int64{1, 2, 3}, 0)
+				if err != nil {
+					return err
+				}
+				err = dbHandler.DBInsertAllReleases(ctx, transaction, "app-1", []int64{1}, 0)
+				if err != nil {
+					return err
+				}
+				err = dbHandler.DBInsertAllReleases(ctx, transaction, "app-2", []int64{1}, 0)
+				if err != nil {
+					return err
+				}
+				err = dbHandler.DBInsertAllReleases(ctx, transaction, "app-3", []int64{1}, 0)
+				if err != nil {
+					return err
+				}
+				err = dbHandler.DBWriteEnvironment(ctx, transaction, "development-1", config.EnvironmentConfig{
+					Upstream: &config.EnvironmentConfigUpstream{
+						Latest: true,
+					},
+				})
+				if err != nil {
+					return err
+				}
+				err = dbHandler.DBWriteEnvironment(ctx, transaction, "development-2", config.EnvironmentConfig{
+					Upstream: &config.EnvironmentConfigUpstream{
+						Latest: true,
+					},
+				})
+				if err != nil {
+					return err
+				}
+				err = dbHandler.DBWriteEnvironment(ctx, transaction, "development-3", config.EnvironmentConfig{
+					Upstream: &config.EnvironmentConfigUpstream{
+						Latest: true,
+					},
+				})
```

Comment:
why are these necessary? We already have the `environmentSetup`, is that not sufficient?

Reply:
The `environmentSetup` only consists of transformers. In order for the transformers to work in the `manifest-repo-export-service`, we need to have some rows in the database(which is actually the `cd-service`'s job).
We had the same approach here:
https://github.com/freiheit-com/kuberpult/blob/231e17bfb039ff752df784f1a8e5a4a4b0414bdb/services/manifest-repo-export-service/pkg/repository/transformer_test.go#L369